src/algorithms/dqns/fpqn.jl

```diff
@@ -0,0 +1,330 @@
+export FPNLearner, FullyParametrizedNet
+
+using Flux
+using CUDA
+using Random
+using Zygote
+using Statistics: mean
+using LinearAlgebra: dot
+
+"""
+    ImplicitQuantileNet(;ψ, ϕ, header)
+
+```
+        quantiles (n_action, n_quantiles, batch_size)
+           ↑
+         header
+           ↑
+feature ↱  ⨀   ↰ transformed embedding
+       ψ       ϕ
+       ↑       ↑
+       s        τ
+```
+"""
+Base.@kwdef struct FullyParametrizedNet{A,B,C} <: AbstractApproximator
```

Comment:
So what's the difference between `FullyParametrizedNet` and `ImplicitQuantileNet`?

Reply:
If so, can we just reuse `ImplicitQuantileNet`?

Reply:
Yes , we can reuse that